src/Background/BackgroundManager.vala

```diff
@@ -123,7 +123,9 @@ namespace Gala {
         }
 
         public void set_size (float width, float height) {
-            background_actor.set_size (width, height);
+            if (width != background_actor.width || height != background_actor.height) {
+                update_background_actor ();
```

Comment:
Wouldn't this trigger the fade animation between the old incorrectly sized background actor and the new one?

We'd probably want to skip the animation in this case.

Reply:
Good point, I considered it not to be important but you're right, it may look odd in the case that a user manages to stumble upon it. Addressed now :)

Reply:
Missing param?
```diff
-update_background_actor ();
+update_background_actor (false);
 ```

Reply:
oh man of course. sorry, good thing you read this carefully. When I tested it I didn't manage to get the animation to show up, so I guess I must have figured that it's disabled.